File: algorithms/2595.number-of-even-and-odd-bits/solution.rb
# @param {Integer} n
# @return {Integer[]}
def even_odd_bit(n)
  even, odd = 0, 0
  n.to_s(2).chars.reverse.each_with_index do |c, i|
    if c == '1'
      if i % 2 == 0
        even += 1
      else
        odd += 1
      end
    end
  end
  [even, odd]
end

File: algorithms/2595.number-of-even-and-odd-bits/solution.rs
impl Solution {
    pub fn even_odd_bit(n: i32) -> Vec<i32> {
        let (mut even, mut odd) = (0, 0);
        let b: String = format!("{:b}", n);
        for (i, c) in b.chars().rev().enumerate() {
            if c == '1' {
                if i % 2 == 0 {
                    even += 1;
                } else {
                    odd += 1;
                }
            }
        }
        vec![even, odd]
    }
}
